test/cpp/end2end/grpc_authz_end2end_test.cc

```diff
@@ -415,6 +433,302 @@ TEST_F(GrpcAuthzEnd2EndTest,
   EXPECT_EQ(resp.message(), kMessage);
 }
 
+TEST_F(GrpcAuthzEnd2EndTest, StaticInitWithAuditLoggingNone) {
+  std::string policy =
+      "{"
+      "  \"name\": \"authz\","
+      "  \"allow_rules\": ["
+      "    {"
+      "      \"name\": \"allow_foo\","
+      "      \"request\": {"
+      "        \"headers\": ["
+      "          {"
+      "            \"key\": \"key-foo\","
+      "            \"values\": [\"foo\"]"
+      "          }"
+      "        ]"
+      "      }"
+      "    }"
+      "  ],"
+      "  \"deny_rules\": ["
+      "    {"
+      "      \"name\": \"deny_bar\","
+      "      \"request\": {"
+      "        \"headers\": ["
+      "          {"
+      "            \"key\": \"key-bar\","
+      "            \"values\": [\"bar\"]"
+      "          }"
+      "        ]"
+      "      }"
+      "    }"
+      "  ],"
+      "  \"audit_logging_options\": {"
+      "    \"audit_condition\": \"NONE\","
+      "    \"audit_loggers\": ["
+      "      {"
+      "        \"name\": \"test_logger\""
+      "      }"
+      "    ]"
+      "  }"
+      "}";
+  InitServer(CreateStaticAuthzPolicyProvider(policy));
+  auto channel = BuildChannel();
+  grpc::testing::EchoResponse resp;
+  grpc::Status status;
+
+  ClientContext context1;
+  // Matches the allow rule.
```

Comment:
Please split each of these cases into their own `TEST_F()` blocks, as per go/tott/649.

Same for all of these tests.

Reply:
Done. Thanks for the link!